src/v/cloud_storage/tests/anomalies_detector_test.cc

```diff
@@ -23,6 +23,19 @@
 #include <boost/test/tools/old/interface.hpp>
 #include <boost/test/unit_test.hpp>
 
+namespace cloud_storage {
+
+bool operator==(const anomalies& lhs, const anomalies& rhs) {
+    return lhs.missing_partition_manifest == rhs.missing_partition_manifest
+           && lhs.missing_spillover_manifests == rhs.missing_spillover_manifests
+           && lhs.missing_segments == rhs.missing_segments
+           && lhs.segment_metadata_anomalies == rhs.segment_metadata_anomalies;
+
+    // anomalies::last_complete_scrub is intentionally omitted
```

Comment:
nit: why does this need to get serialized as a part of this struct anyway? Could we just maintain a max counter that gets updated in `process_anomalies()`?

Reply:
There's no functional reason really. I was trying to keep the number of new fields added to the manifest low, so I decided to bundle this field into the `anomalies` struct in order to avoid adding it directly to the manifest. It's also convenient for reporting, since we just grab the `anomalies` struct and convert it to JSON format.

I don't feel too strongly either way. If you think a different approach is preferable, I can change the code.

Reply:
Np, I don't feel too strongly about it. In general I think it'll be less error prone / surprising to lean towards encapsulating the needs-to-be-serialized stuff as an envelope, and then adding metadata for logging in some parent struct (admittedly, would be slightly more verbose/annoying, but it sounds nice in theory).